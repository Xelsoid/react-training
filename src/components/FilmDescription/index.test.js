import React from 'react';
import MOCKED_DATA from '@mockedData/MOCKED_DATA';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import ComponentExample from './index';

const mockStore = configureStore([]);

describe('FilmDescription test', () => {
  let store;
  const match = {
    params: {
      id: '337167',
    },
  };

  beforeEach(() => {
    store = mockStore({
      moviesData: MOCKED_DATA,
    });
  });

  it('renders correctly', () => {
    const component = mount(
      <Provider store={store}>
        <ComponentExample match={match} />
      </Provider>,
    );
    expect(component).toMatchSnapshot();
  });
});
