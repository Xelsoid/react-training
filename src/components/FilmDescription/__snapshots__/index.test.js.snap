// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmDescription test renders correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <FilmDescription
    match={
      Object {
        "params": Object {
          "id": "337167",
        },
      }
    }
    overview="title"
    poster_path="title"
    release_date="title"
    runtime={0}
    tagline="title"
    title="title"
    vote_average={0}
  >
    <div
      className="film-description"
    >
      <div
        className="film-description__image-wrapper"
      >
        <img
          alt="title"
          className="film-description__image"
          onError={[Function]}
          src="title"
        />
      </div>
      <div
        className="film-description__info-wrapper"
      >
        <div
          className="film-description__title-wrapper"
        >
          <h2
            className="film-description__title"
          >
            title
          </h2>
          <div
            className="film-description__rating"
          >
            0
          </div>
        </div>
        <small
          className="film-description__tagline"
        >
          title
        </small>
        <div
          className="film-description__year-wrapper"
        >
          <span
            className="film-description__year"
          >
            titl
          </span>
          <span
            className="film-description__runtime"
          >
            0 min
          </span>
        </div>
        <p
          className="film-description__narration"
        >
          title
        </p>
      </div>
    </div>
  </FilmDescription>
</Provider>
`;
